decidim/decidim
# Add meetings documents as collections in participatory spaces landing pages

#### :tophat: What? Why?

This PR:
* Defines a ComponentAttachmentCollectionPresenter class to present the documents of a components as an attachment collection
* Allows to include in documents panel cell additional documents collections of components and uses it in participatory spaces documents content blocks with the previously described presenter for published meeting components. Each component documents collection is shown in a folder named with the name of the component
* Adds some system tests and some more for the presenter

#### :pushpin: Related Issues

- Fixes #12624

#### Testing

* As admin:
  * Publish a meetings component in a participatory space, publish a meeting on it and attach a document (it can be inside a folder too)
  * Add the related documents content block to the landing page of the participatory space
* As any user visit the landing page of the participatory space. A folder named as "_component name_ Documents" should include the previously defined attachment with the rest of documents attached to the other meetings published in the component if present

The feature can be checked at https://decidim-lot2.populate.tools/processes/polls-in-meetings. The example process has a component with a meeting and a [PDF document attached](https://decidim-lot2.populate.tools/admin/participatory_processes/polls-in-meetings/components/3/manage/meetings/3/attachments)

### :camera: Screenshots

[meetings_documents.webm](https://github.com/decidim/decidim/assets/446459/966e8121-57a1-4896-a35b-193ea282a0b0)
